### ISSUE TITLE
[Reactant] Retire O(K) batched gather workaround when minimum Reactant is ≥ 0.2.284

### ISSUE THREAD

**Issue by nsiccha** [nsiccha/ReactiveKernels.jl#2]
## Summary

ReactiveKernels uses an O(K) mask/select reduction for a traced vector gather inside Reactant's internal `Ops.batch` surface. This works around a silent Reactant 0.2.278 miscompile in which every lane uses lane 1's dynamic index.

The direct indexed spelling is correct in Reactant 0.2.284, so RK should retire the workaround once its supported Reactant floor and receipts permit it.

`Reactant.Ops.batch` is internal: it is not exported, documented, or marked public. RK's two call sites do match its implemented leading-batch-axis contract; this issue tracks an RK dependency/workaround boundary rather than claiming a public Reactant API regression.

## Minimal reproducer

```julia
using Reactant

Reactant.set_default_backend("cpu")

function lane_getindex(v, i)
    idx = Reactant.@allowscalar i[]
    Reactant.@allowscalar v[idx]
end

gather(V, I) = only(Reactant.Ops.batch(
    lane_getindex,
    Reactant.TracedRArray[V, I],
    Int64[size(V, 1)],
))

V_host = reshape(collect(1.0:12.0), 4, 3)
I_host = [1, 3, 2, 3]
V = Reactant.to_rarray(V_host)
I = Reactant.to_rarray(I_host)

@show Array((@compile gather(V, I))(V, I))
@show [V_host[k, I_host[k]] for k in 1:4]
```

Expected: `[1.0, 10.0, 7.0, 12.0]`.

- Reactant 0.2.278: `[1.0, 2.0, 3.0, 4.0]`
- Reactant 0.2.284: `[1.0, 10.0, 7.0, 12.0]`

## Current RK workaround

`ext/ReactiveKernelsReactantExt.jl` implements traced one-dimensional indexing as:

```julia
sum(ifelse.(collect(eachindex(array)) .== index, array, zero(T)))
```

That spelling is exact and differentiable, but O(K) instead of O(1). It was introduced in commit `4d35b7948888d89c6350059dc7163fb7b89eb548` after `plate(eachcol(logits), y)` selected every lane's first label.

## Exit criteria

- [ ] Raise or otherwise establish the tested Reactant floor at 0.2.284 or newer.
- [ ] Replace the mask/select reduction with direct traced indexing.
- [ ] Keep the lane-varying gather test in `test/test_mnist_full_joint_reactant.jl` as the regression guard.
- [ ] Regenerate affected Reactant benchmark receipts and verify MNIST likelihood parity.

Tested with Julia 1.10.11, Reactant 0.2.278 (broken) and 0.2.284 (fixed), ReactantCore 0.1.21, CPU backend.

**Comment by nsiccha** [comment]
Fixed by canonical merge 7deed8c28658ee2c0bba6cbaeefe8763cbedc283: ReactiveKernels now requires Reactant 0.2.284 and uses direct traced gather instead of the O(K) mask/select workaround. Hosted Reactant integration passed on canonical descendant 3232935588dd8434171fa8fa9f862bffbaa13a36 in run 33970269989, and docs run 33970269982 passed. The broad run ended after 90 minutes with 12 of 18 jobs successful and no failures; six core jobs were cancelled while in test_mutation_profile_b.jl. The same core-suite hang is present on pre-fix baseline 71c911b8e858a682b7acb6decdfc97bd599d333c (run 33958386461), so it is tracked independently rather than attributed to this issue.
